cmd/csi-attacher/main.go

```diff
@@ -134,8 +135,17 @@ func main() {
 		os.Exit(1)
 	}
 	klog.V(2).Infof("CSI driver name: %q", csiAttacher)
-	metricsManager.SetDriverName(csiAttacher)
-	metricsManager.StartMetricsEndpoint(*metricsAddress, *metricsPath)
+
+	// Prepare http endpoint for metrics + leader election healthz
+	mux := http.NewServeMux()
+	cmm := metrics.NewCSIMetricsManagerForSidecar(csiAttacher)
```

Comment:
While copying this code into external-health-monitor I noticed that `metricsPath` is still allowed to be empty, which means that an if check for that is missing here.

Also, metrics gathering elsewhere instantiates the metrics manager *before* starting to talk to the sidecar and then changes the driver name later - not sure how important that is.

Reply:
Actually... I think the metrics data is now always going to be empty because the HTTP server presents data from the `cmm` instance and not the `metricsManager` instance that is used to monitor CSI calls.

Reply:
Here's my take on the same update in external-health-monitor: https://github.com/kubernetes-csi/external-health-monitor/pull/46